zip-0244.rst

```diff
@@ -1,8 +1,636 @@
+   T.4b.iii: sapling_outputs_noncompact_digest (32-byte hash)
+
+The personalization field of this hash is set to::
+
+  "ZTxIdSOutputHash"
+
+T.4b.i: sapling_outputs_compact_digest
+......................................
+A BLAKE2b-256 hash of the subset of Sapling output information included in the
+ZIP-307 [#zip-0307]_ ``CompactBlock`` format for all Sapling shielded outputs
+belonging to the transaction. For each output, the following elements are included
+in the hash::
+
+   T.4b.i.1: cmu                  (field encoding bytes)
+   T.4b.i.2: ephemeral_key        (field encoding bytes)
+   T.4b.i.3: enc_ciphertext[..52] (First 52 bytes of field encoding)
+
+The personalization field of this hash is set to::
+
+  "ZTxIdSOutC__Hash" (2 underscore characters)
+
+T.4a.ii: sapling_outputs_memos_digest
+.....................................
+A BLAKE2b-256 hash of the subset of Sapling shielded memo field data for all Sapling
+shielded outputs belonging to the transaction. For each output, the following elements
+are included in the hash::
+
+   T.4b.ii.1: enc_ciphertext[52..564] (contents of the encrypted memo field)
+
+The personalization field of this hash is set to::
+
+  "ZTxIdSOutM__Hash" (2 underscore characters)
+
+T.4a.iii: sapling_outputs_noncompact_digest
+...........................................
+A BLAKE2b-256 hash of the remaining subset of Sapling output information **not** included
+in the ZIP 307 [#zip-0307]_ ``CompactBlock`` format, excluding zkproof data, for all
+Sapling shielded outputs belonging to the transaction. For each output, the following
+elements are included in the hash::
+
+   T.4b.iii.1: cv                    (field encoding bytes)
+   T.4b.iii.2: enc_ciphertext[564..] (post-memo suffix of field encoding)
+   T.4b.iii.3: out_ciphertext        (field encoding bytes)
+
+The personalization field of this hash is set to::
+
+  "ZTxIdSOutN__Hash" (2 underscore characters)
+
+Signature Digest
+================
+
+A new per-input transaction digest algorithm is defined that constructs a hash that may be
+signed by a transaction creator to commit to the effects of the transaction. In the case
+that the transaction consumes no transparent inputs, it should be possible to just sign
+the transaction identifier produced by the ``TxId Digest`` algorithm.  In the case that
+transparent inputs are present, this algorithm follows closely the ZIP 143 [#zip-0143]_
+algorithm.
+
+The overall structure of the hash is as follows; each name referenced here will be
+described in detail below::
+
+    signature_digest
+    ├── header_digest
+    ├── transparent_digest
+    ├── sprout_digest
+    └── sapling_digest
+
+signature_digest
+----------------
+A BLAKE2b-256 hash of the following values ::
+
+   S.1: header_digest      (32-byte hash output)
+   S.2: transparent_digest (32-byte hash output)
+   S.3: sprout_digest      (32-byte hash output)
+   S.4: sapling_digest     (32-byte hash output)
```

Comment:
These fields have the same names as the ones in the `txid_digest`, which is slightly confusing because this `transparent_digest` is not the same, in general. Should we use `*_sig_digest` for the fields that can potentially be different?

Reply:
Maybe for `transparent_digest` alone it should have a different name? The other three are the same as for the txid digest.